app/ide-desktop/lib/cloud/src/authentication/src/api.tsx

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file Module containing the API client for the Cloud backend API.
+ * 
+ * Each exported function in this module corresponds to an API endpoint. The functions are
+ * asynchronous and return a `Promise` that resolves to the response from the API.
+ */
+import { API_URL } from "./config";
+
+
+
+// =============
+// === Types ===
+// =============
+
+interface Request {
+    accessToken: string;
+    path: string;
+    method: HttpMethod;
+    body?: Record<string, string>;
+}
+
+enum HttpMethod {
+    get = "GET",
+    post = "POST",
+    put = "PUT",
+    delete = "DELETE",
+}
+
+export type Organization = {
+    id: string;
+    userEmail: string;
+    name: string;
+}
+
+
+
+// ================
+// === FetchApi ===
+// ================
+
+/// Utility function for performing a REST request to an API endpoint.
+///
+/// Parses the response as JSON (if possible) and returns the result.
+/// Otherwise returns `null` on a successful API call.
+const fetchApi = async (
+    {
+        accessToken,
+        path,
+        method,
+        body,
+    }: Request,
+): Promise<Response> => {
+    if (!accessToken) { throw new Error("No access token provided"); }
+    
+    const config: RequestInit = { method };
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const headers = new Headers(config.headers);
+    headers.set("Authorization", `Bearer ${accessToken}`);
+    if (body != null) {
+        config.body = JSON.stringify(body);
+        headers.set("Content-Type", "application/json");
+    }
+
+    config.headers = headers;
+    const response = await fetch(`${API_URL}/${path}`, config);
+
+    return response;
+}
+
+
+
+// ===================
+// === SetUsername ===
+// ===================
+
+export type SetUsernameBody = {
+    userName: string;
+    userEmail: string;
+};
```

Comment:
1. 👍 
2. The fields must have the exact names they do because that's what we deserialize to, so we *do* need to repeat the names. As to why they're repeated on the backend, that can be changed but it's out of scope for this PR I think.
3. 👍 
4. The field stores the username and it's `userName` and the struct is for the HTTP body of a request to the `/set-username` endpoint so I'm not sure what else to call it. I get that it's repetitive but the fact that we provide the username to an endpoint that sets the username is going to inherently have duplication, no?